server/task/src/test/java/org/apache/james/task/eventsourcing/TerminationSubscriberContract.java

```diff
@@ -32,86 +33,112 @@
     }
 
     @Test
     default void handlingNonTerminalEventShouldNotBeListed() {
         TerminationSubscriber subscriber = subscriber();
         TaskEvent event = new Started(new TaskAggregateId(TaskId.generateTaskId()), EventId.fromSerialized(42), new Hostname("foo"));
 
-        subscriber.handle(event);
+        sendEvents(subscriber, event);
 
         assertEvents(subscriber).isEmpty();
     }
 
     @Test
     default void handlingMultipleEventsShouldBeListed() {
         TerminationSubscriber subscriber = subscriber();
-        TaskEvent event1 = new Completed(new TaskAggregateId(TaskId.generateTaskId()), EventId.fromSerialized(42), Task.Result.COMPLETED);
-        TaskEvent event2 = new Failed(new TaskAggregateId(TaskId.generateTaskId()), EventId.fromSerialized(42));
-        TaskEvent event3 = new Cancelled(new TaskAggregateId(TaskId.generateTaskId()), EventId.fromSerialized(42));
 
-        subscriber.handle(event1);
-        subscriber.handle(event2);
-        subscriber.handle(event3);
+        sendEvents(subscriber, COMPLETED_EVENT, FAILED_EVENT, CANCELLED_EVENT);
 
-        assertEvents(subscriber).containsExactly(event1, event2, event3);
+        assertEvents(subscriber).containsExactly(COMPLETED_EVENT, FAILED_EVENT, CANCELLED_EVENT);
     }
 
     @Test
-    default void listeningEventsShouldBeContinuous() {
+    default void multipleListeningEventsShouldShareEvents() {
         TerminationSubscriber subscriber = subscriber();
-        TaskEvent event1 = new Completed(new TaskAggregateId(TaskId.generateTaskId()), EventId.fromSerialized(42), Task.Result.COMPLETED);
-        TaskEvent event2 = new Failed(new TaskAggregateId(TaskId.generateTaskId()), EventId.fromSerialized(42));
-        TaskEvent event3 = new Cancelled(new TaskAggregateId(TaskId.generateTaskId()), EventId.fromSerialized(42));
 
-        Flux.just(event1, event2, event3)
+        sendEvents(subscriber, COMPLETED_EVENT, FAILED_EVENT, CANCELLED_EVENT);
+
+        List<List<Event>> listenedEvents = Flux.range(0, 2)
             .subscribeOn(Schedulers.elastic())
-            .delayElements(Duration.ofMillis(20))
-            .doOnNext(subscriber::handle)
-            .subscribe();
+            .flatMap(ignored -> collectEvents(subscriber))
+            .collectList()
+            .block();
+        assertThat(listenedEvents).hasSize(2);
+        assertThat(listenedEvents.get(0)).containsExactly(COMPLETED_EVENT, FAILED_EVENT, CANCELLED_EVENT);
+        assertThat(listenedEvents.get(1)).isEqualTo(listenedEvents.get(0));
+    }
+
+    @Test
+    default void dynamicListeningEventsShouldGetOnlyNewEvents() {
+        TerminationSubscriber subscriber = subscriber();
+
+        sendEvents(subscriber, COMPLETED_EVENT, FAILED_EVENT, CANCELLED_EVENT);
 
-        assertEvents(subscriber).containsExactly(event1, event2, event3);
+        List<Event> listenedEvents = Mono.delay(Duration.ofMillis(Math.round(DELAY_BEFORE_PUBLISHING + 1.5 * DELAY_BETWEEN_EVENTS)))
```

Comment:
if you can live without the non-integer part of the multiplication, Duration allow to do the maths for this